scene/animation/animation_mixer.cpp

```diff
@@ -2128,6 +2128,11 @@ void AnimationMixer::_bind_methods() {
 	ADD_SIGNAL(MethodInfo(SNAME("animation_finished"), PropertyInfo(Variant::STRING_NAME, "anim_name")));
 	ADD_SIGNAL(MethodInfo(SNAME("animation_started"), PropertyInfo(Variant::STRING_NAME, "anim_name")));
 	ADD_SIGNAL(MethodInfo(SNAME("caches_cleared")));
+
+#ifdef TOOLS_ENABLED
```

Comment:
The API should be the same for editor and runtime, so this shouldn't be in `TOOLS_ENABLED` but always exposed.

Reply:
Before AnimationMixer was implemented, this was an API that was only valid for TOOLS_ENABLED. I assume that as long as the prefix _ is there it won't actually be exposed, but is there a case for using it at runtime?

Reply:
I don't expect users to use these methods at runtime, especially since they're prefixed with `_` so they're hidden. But for classes which are part of the public API, we try to make sure we don't create a discrepancy between the API exposed in `TOOLS_ENABLED` and non-tools.

Here again the `_` prefix will hide it, but if that were to change in the future, we shouldn't have methods that count in the API hash / GDExtension API that actually won't exist at runtime, but aren't documented as editor-only.

I see in 4.1 the `restore` method was only for `TOOLS_ENABLED`, but it was also exposed in `AnimatedValuesBackup` which isn't part of the public (scripting) API.

So I don't have a concrete use case in mind, it's more a code consistency guideline that the API we expose in `TOOLS_ENABLED` and non `TOOLS_ENABLED` should be the same for core classes.

Reply:
Exposed functions except `apply_reset()` since `apply_reset()` depends on the editor feature such as UndoRedo and dummy player.